### PR TITLE
One way arrow on roads

### LINKED ISSUE #77: One way roads
I noticed that `oneway` is an attribute on the transportation class of openmaptiles, but from a search of the repository it doesn't look like it's used. It might be nice to add a subtle arrow on one way roads

### PR DESCRIPTION
**Pull request by kylebarron**
Closes #77.

Preview: https://maputnik.github.io/editor/?style=https://raw.githubusercontent.com/kylebarron/osm-liberty/one-way/style.json

Currently set to a `minzoom` of 15, which seems fine. On merge, the `sprite` key needs to be set back to the default. 

Note that in the [openmaptiles schema](https://openmaptiles.org/schema/#transportation), `oneway` can have a value of `1`, `-1`, or `0`. Where `1` is one way going in the same direction as the line, 0 is not a one way, and `-1` is a one way in the opposite direction. Currently only `1` is supported. I don't know how prevalent `-1` is, but to be complete, it would be good to add a backwards arrow to use on those occasions.

[image](https://maputnik.github.io/editor/?style=https://raw.githubusercontent.com/kylebarron/osm-liberty/one-way/style.json#15.29/42.37176/-71.111299)

image

### CODE REVIEW

**Comment by orangemug** [comment]
Looks good, one question what would happen to the arrow if the corner in the middle of this image was tighter, would it overflow or would mapbox-gl-js cull it?

Screenshot 2020-03-09 at 20 07 24

**Comment by kylebarron** [comment]
I'm not sure. I don't think there's anything keeping the arrow inside the line.

If I take off the road lines, the arrows still display:
image

But maybe Mapbox automatically prevents displaying the symbols when the line is very curvy. Or maybe I'm missing an option.

**Comment by pathmapper** [comment]
> I don't know how prevalent -1 is

See https://taginfo.openstreetmap.org/keys/oneway#values

> but to be complete, it would be good to add a backwards arrow to use on those occasions.

We could use the same arrow for this:
Just add a second layer which is filtering for `oneway = -1`. For this layer use the same arrow, but with `"icon-rotate": 180`. @kylebarron would you mind adding this to the PR?

File: style.json [hunk 1]
```diff
@@ -893,6 +893,20 @@
         "line-width": {"base": 1.2, "stops": [[13.5, 0], [14, 2.5], [20, 18]]}
       }
     },
+    {
+      "id": "road_one_way_arrow",
+      "type": "symbol",
+      "source": "openmaptiles",
+      "source-layer": "transportation",
+      "minzoom": 15,
+      "filter": ["all", ["in", "oneway", 1]],
```
Comment (pathmapper) [inline style.json]: This is simpler:

```suggestion
      "filter": ["==", "oneway", 1],
```

File: style.json [hunk 2]
```diff
@@ -893,6 +893,20 @@
...
+      "layout": {
+        "icon-image": "arrow",
+        "symbol-placement": "line",
+        "symbol-spacing": 250,
```
Comment (pathmapper) [inline style.json]: This is the default value and could be removed:

```suggestion
```

File: style.json [hunk 3]
```diff
@@ -893,6 +893,20 @@
...
+        "icon-size": 1
```
Comment (pathmapper) [inline style.json]: Also the default value and could be removed:

```suggestion
```

**Review by pathmapper — CHANGES_REQUESTED** [review]
@kylebarron left some feedback, otherwise it looks good.

**Comment by kylebarron** [comment]
Ok, I added the one way arrow for `oneway=-1`. If this looks good to go, I'll change the `sprite` url back to the master branch.

**Review by pathmapper — APPROVED** [review]

**Comment by pathmapper** [comment]
Thanks @kylebarron, looks good to merge.

Yes, please change the sprite url back to the `gh-pages` branch (I think this is what you've meant, there is no `master` branch for historical reasons :smile: ).

**Comment by kylebarron** [comment]
The `sprite` key is reset now

**Pull request merged**